p2p/trust/trustmetric.go

```diff
@@ -0,0 +1,394 @@
+package trust
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	store *trustMetricStore
+)
+
+type peerMetricRequest struct {
+	Key  string
+	Resp chan *TrustMetric
+}
+
+type trustMetricStore struct {
+	PeerMetrics map[string]*TrustMetric
+	Requests    chan *peerMetricRequest
```

Comment:
do these channels need to be public? And can we get comments on them please? Same with all the other structs and functions.